pyvista/themes.py

```diff
@@ -779,75 +1103,74 @@ def set_to_document(self):
         self._axes['y_color'] = 'seagreen'
         self._axes['z_color'] = 'blue'
 
-    def _set_to_testing(self):
-        """Low resolution testing theme.
 
-        Necessary for image regression.  Xvfb doesn't support
-        multi-sampling, so we disable it here for consistency between
-        desktops and remote testing.
-        """
-        self.restore_defaults()
-        self._off_screen = True
+class _TestingTheme(Theme):
+    """Low resolution testing theme for ``pytest``.
+
+    Necessary for image regression.  Xvfb doesn't support
+    multi-sampling, so we disable it here for consistency between
+    desktops and remote testing.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._name = 'testing'
         self._multi_samples = 1
         self._window_size = [400, 400]
 
-    def set_to_dark(self):
-        """Set the global theme to dark mode.
 
-        Black background, "viridis" colormap, tan meshes, white (hidden) edges.
+class Defaults(Theme):
+    """Global PyVista defaults.
+
+    Examples
+    --------
+    Change the default background color to white.
+
+    >>> import pyvista
+    >>> pyvista.defaults.color = 'white'
+
+    Show edges by default.
+
+    >>> pyvista.defaults.show_edges = True
+
+    """
+
+    def __init__(self):
+        """Initialize the base theme."""
+        super().__init__()
+
+    def load_theme(self, theme):
```

Comment:
Hmm, why not put this functionallity inside `class Theme` Usecase - if I want to use the current default-settings as a base for `my_theme` I don't know how to do it without copying everything by hand. Concret usecase (which has to be discussed further) would be inside `BasePlotter.__init__()`:

```
        if theme is None:
            self._theme = pyvista.defaults

# maybe should be changed to something like:
        if theme is None:
            self._theme = pyvista.Theme().load_theme(pyvista.defaults)
```
The reasoning behind the change would be a clear behavior. A Plotter uses the setting from when it was created. Otherwise the (imho unwanted) mixed behavior appears, where a change to pyvista.defaults after Plotter-creation might change plot-results for some cases (e.g. add_mesh()), but would not take effect for others (e.g. background). The other option - to "recreate" all plotter-objects to make all default-setting take effect whenever pyvista.defaults is changed seems not feasible to me.

Reply:
> self._theme = pyvista.Theme().load_theme(pyvista.defaults)

That's a really good idea!  That way there's no side effect when changing the default global theme.